### DISCUSSION TITLE
Using open source llms with agency swarm

### DISCUSSION THREAD

**Discussion by hamidg20032** [VRSEN/agency-swarm#61] [Q&A]
is it possible to use open source llms with agency swarm through ollama, if not can you please implement this feature

**Comment by Tedfulk** [comment]
Idk, it seems open source models don't have the capability to do function calling. I think we gotta wait the models get better. Gpt-4 still king.

**Comment by hamidg20032** [comment]
was hoping at least adding the base_url function to agency swarm so we have the chance to try to use open source llms by putting a local url (like llm studio or janAI)

**Comment by cbruyndoncx** [comment]
There is an OpenOpenAI github repo to host the OpenAI assistants API, so as far as i can tell, locally hosting is feasible if you put in the work.

**Comment by nicko-ai** [comment]
This discussion is outdated, as we released open-source model support a while ago:
https://agency-swarm.ai/additional-features/open-source-models

In the new framework version based on the Responses API, you’ll have full control over your conversation history and built-in support for any LLMs.
